h2root fragilities

### Check duplicate issues.

- [x] Checked for duplicates

### Description

Before digging further on this, please take a look at https://github.com/root-project/root/pull/19236#issuecomment-3023003515 in case it's a won't fix, or in case it takes less time to rewrite from scratch in C++ and forget about gfortran. That would also avoid the issue that it does not work on Windows.

This is a collection of memory corruption issues / warnings / etc. probably leading to the fact that it crashes in some platforms if -O3 flag is activated, or even with -O0 on mac13/mac14, etc. These were scattered across several PRs and issues, so just joining all info in a single place.

```
/opt/root_src/misc/minicern/src/zebra.f:6743:72: runtime error: load of address 0x6550d1529b4c with insufficient space for an object of type 'integer(kind=4)'
0x6550d1529b4c: note: pointer points here
  24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
              ^ 
/opt/root_src/misc/minicern/src/zebra.f:6744:72: runtime error: load of address 0x6550d1529b4c with insufficient space for an object of type 'integer(kind=4)'
0x6550d1529b4c: note: pointer points here
  24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
              ^ 
/opt/root_src/misc/minicern/src/zebra.f:6750:72: runtime error: store to address 0x6550d1529b4c with insufficient space for an object of type 'integer(kind=4)'
0x6550d1529b4c: note: pointer points here
  24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
              ^ 
```

which I am not sure how to solve.

Aside:
On Windows, bellenot found these warnings: https://github.com/user-attachments/files/20980738/ifx-logs.zip

such as
```
C:\root-dev\git\master\misc\minicern\src\zebra.f(6649): warning #6717: This name has not been given an explicit type.   [LQ]
      COMMON /ZEBQ/ IQFENC(4), LQ(100)
```


And in the past I also got this one:

```
  ==84992==ERROR: AddressSanitizer: memcpy-param-overlap: memory ranges [0xffffd9c19d70,0xfffed9c19d72) and [0x0000004d2160, 0xffffffff004d2162) overlap
      #0 0xffff95246d08 in __interceptor_memcpy (/lib64/libasan.so.6+0x3ad08)
      #1 0x43ff88 in hropen_ /github/home/ROOT-CI/src/misc/minicern/src/hbook.f:251
      #2 0x405ae8 in main /github/home/ROOT-CI/src/main/src/h2root.cxx:315
      #3 0xffff9367927c in __libc_start_call_main (/lib64/libc.so.6+0x2727c)
      #4 0xffff93679354 in __libc_start_main_alias_2 (/lib64/libc.so.6+0x27354)
      #5 0x405e6c in _start (/github/home/ROOT-CI/build/bin/h2root+0x405e6c)
  
  Address 0xffffd9c19d70 is located in stack of thread T0 at offset 48 in frame
      #0 0x43f81c in hropen_ /github/home/ROOT-CI/src/misc/minicern/src/hbook.f:239
  
    This frame has 5 object(s):
      [48, 56) 'chopt' (line 250) <== Memory access at offset 48 partially overflows this variable
      [80, 144) 'close_parm.62' (line 281) <== Memory access at offset 48 partially underflows this variable
      [176, 704) 'dt_parm.57' (line 255) <== Memory access at offset 48 partially underflows this variable
      [832, 1360) 'dt_parm.59' (line 266) <== Memory access at offset 48 partially underflows this variable
      [1488, 2016) 'dt_parm.61' (line 279) <== Memory access at offset 48 partially underflows this variable
  HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
        (longjmp and C++ exceptions *are* supported)
  Address 0x0000004d2160 is a wild pointer.
  SUMMARY: AddressSanitizer: memcpy-param-overlap (/lib64/libasan.so.6+0x3ad08) in __interceptor_memcpy
  ==84992==ABORTING
```

And if I unsilence the compile warnings (https://github.com/root-project/root/commit/d8215a8a898d1dca1373dccbd0eef40e9a5b5f99) I get:

```
 4023 |       CALL HRZIN(IHWORK,0,0,KEYS,9999,'SC')
      |                        1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:4028:24:

 4028 |       CALL HRZIN(IHWORK,0,0,KEYS,9999,'SNC')
      |                        1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:4040:24:

 4040 |       CALL HRZIN(IHWORK,LHWORK,1,KEYS,9999,'SND')
      |                        1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:454:26:

  454 |          CALL HRZIN(IHDIV,0,0,KEYS,9999,'SC')
      |                          1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:475:23:

  475 |       CALL HRZIN(IHDIV,0,0,KEYS,ICYCLE,'NC')
      |                       1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:496:26:

  496 |          CALL HRZIN(IHDIV,LCDIR,-2,KEYS,ICYCLE,'ND')
      |                          1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:506:26:

  506 |          CALL HRZIN(IHDIV,LLID,  0,KEYS,ICYCLE,'ND')
      |                          1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:2184:17:

 1590 |          CALL VZERO(IFLAG,31)
      |                    2
......
 2184 |       CALL VZERO(I1,37)
      |                 1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:2198:24:

 2058 |             CALL UCTOH(CHPAT(IL),IHDIR,4,16)
      |                                 2
......
 2198 |       CALL UCTOH('NO  ',INO,4,4)
      |                        1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:2200:24:

 2058 |             CALL UCTOH(CHPAT(IL),IHDIR,4,16)
      |                                 2
......
 2200 |       CALL UCTOH('$   ',IDOL,4,4)
      |                        1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:3159:32:

 3159 |                CALL HRZIN(IHDIV,LBUF,-LCIND,KEYS,99999,'R')
      |                                1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:3163:29:

 3163 |             CALL HRZIN(IHDIV,LBUF,-LCIND,KEYS,99999,'R')
      |                             1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:3167:32:

 3167 |                CALL HRZIN(IHDIV,LBUF,-LCIND,KEYS,99999,'R')
      |                                1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:3044:29:

 3044 |             CALL HRZIN(IHDIV,0,0,KEYS,ICYCLE,'C')
      |                             1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:3053:29:

 3053 |             CALL HRZIN(IHDIV,LBUF,-LCIND,KEYS,ICYCLE,' ')
      |                             1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:1040:32:

 1040 |                CALL HRZIN(IHDIV,LCIDN,-1,KEYS,99999,'RS')
      |                                1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:1049:32:

 1049 |                CALL HRZIN(IHDIV,LCIDN,-1,KEYS,99999,'R')
      |                                1
Warning: Rank mismatch in argument ‘lbank’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/hbook.f:1074:32:

 1074 |       CALL HGNT1(IDN, '*', '*', 0, 0, IDNEVT, IERROR)
      |                                1
Warning: Rank mismatch in argument ‘ioffst’ at (1) (rank-1 and scalar)
[10/15 2.8/sec] Building Fortran object misc/minicern/CMakeFiles/minicern.dir/src/zebra.f.o
/opt/root_src/misc/minicern/src/zebra.f:4320:52:

 4320 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR) CALL MZSDIV (IXSTOR,-7)
      |                                                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4324:19:

 4324 |       CALL MZCHLS (-7,LHEAD)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4332:26:

 4332 |       CALL MZFLAG (IXSTOR,LHEAD,IQDROP,'V')
      |                          1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4335:26:

 4335 |    31 CALL MZFLAG (IXSTOR,LHEAD,IQDROP,'.')
      |                          1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4339:19:

 4339 |       CALL MZCHLS (-7,LN)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4344:26:

 4344 |    41 CALL MZFLAG (IXSTOR,LHEAD,IQDROP,'L')
      |                          1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4482:23:

 4482 |       MQDVGA = MZDVAC (IXGARB)
      |                       1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4485:23:

 4485 |       MQDVWI = MZDVAC (IXWIPE)
      |                       1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4488:23:

 4488 |    16 MQDVWI = MZDVAC (IXWIPE)
      |                       1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2733:34:

 2733 |                CALL MZDROP(JQPDVS,LBANK,' ')
      |                                  1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1890:19:

 1890 |    22 CALL MZSDIV (JDV,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1903:19:

 1903 |       CALL MZCHLS (-7,LQSUP)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1920:19:

 1920 |       CALL MZCHLS (-7,LNEXT)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1945:36:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2        
......
 1945 |           CALL UCOPYI (LQ(KQS+IQLN),NQIOCH,NQNIO+1)
      |                                    1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:1986:40:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2            
......
 1986 |       CALL UCOPYI (IQ(KQSP+LIOD+IXIO+1),NQIOCH,NQNIO+1)
      |                                        1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:2304:25:

 2304 |       CALL MZSDIV (IXDIV,0)
      |                         1
Warning: Rank mismatch in argument ‘iflagp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2313:23:

 2313 |       JQDIVI = MZFDIV (-7, LORG)
      |                       1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2315:19:

 2315 |       CALL MZCHLS (-7,LORG)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2350:30:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2  
......
 2350 |       CALL UCOPYI (LQ(KQS+LN),NQIOCH,NQNIO+1)
      |                              1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:2368:19:

 2368 |       CALL MZPUDX (LN,NWD)
      |                   1
Warning: Rank mismatch in argument ‘lp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2399:19:

 2399 |       CALL MZPUDX (L,NWD)
      |                   1
Warning: Rank mismatch in argument ‘lp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2430:19:

 2430 |       CALL MZPUDX (LN,-INCNL)
      |                   1
Warning: Rank mismatch in argument ‘lp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2435:19:

 2435 |       CALL MZPUDX (LD,NWD)
      |                   1
Warning: Rank mismatch in argument ‘lp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2443:19:

 2443 |       CALL MZLIFT (-7,LNEW,0,63,NQID,-1)
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5627:19:

 5627 |       CALL IZBCDT (NCH,ITAB)
      |                   1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5709:19:

 5709 |       CALL MZIOCF (0,MXVALA)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5715:19:

 5715 |       CALL MZIOCF (0,MXVALB)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5742:19:

 5742 |       CALL MZIOCF (0,MXVALA)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5765:19:

 5765 |       CALL MZIOCF (0,MXVALA)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5773:19:

 5773 |       CALL MZIOCF (0,MXVALB)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5788:19:

 5788 |       CALL MZIOCF (0,MXVALB)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5807:19:

 5807 |       CALL MZIOCF (JU,MXVALC)
      |                   1
Warning: Rank mismatch in argument ‘jup’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5830:30:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2  
......
 5830 |       CALL UCOPYI (IQUEST(12),IODVEC,NWIO+1)
      |                              1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:1807:28:

 1807 |           CALL UCOPYI (NIOP,NQIOCH,NIO+1)
      |                            1
......
 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:1810:38:

 1810 |         CALL MZLIFT (IXP,LP,LSUPP,63, NQID, NZP)
      |                                      1
Warning: Rank mismatch in argument ‘name’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:799:32:

  799 |             CALL UCOPYI(IQ(KOF),IBUF,JREC)
      |                                1
......
 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2    
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:2153:53:

 2153 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR)  CALL MZSDIV (IXSTOR,-7)
      |                                                     1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2159:23:

 2159 |           CALL MZPUSH (-7,LSYS,0,100,'I')
      |                       1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2177:38:

 1797 |       IF  (NIO.NE.0)  CALL UCTOH (CHIDH,NQID,4,NIO)
      |                                        2
......
 2177 |       IF (N.NE.0)  CALL UCTOH (CHNAME,NAME,4,N)
      |                                      1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:3099:26:

 1797 |       IF  (NIO.NE.0)  CALL UCTOH (CHIDH,NQID,4,NIO)
      |                                        2
......
 3099 |       CALL UCTOH(CHPAT(1),IHDIR,4,16)
      |                          1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:3100:17:

 3100 |       CALL ZHTOI(IHDIR,IHDIR,4)
      |                 1
Warning: Actual argument contains too few elements for dummy argument ‘holl’ (4/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:3123:30:

 1797 |       IF  (NIO.NE.0)  CALL UCTOH (CHIDH,NQID,4,NIO)
      |                                        2
......
 3123 |          CALL UCTOH(CHPAT(IL),IHDIR,4,16)
      |                              1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:3124:20:

 3124 |          CALL ZHTOI(IHDIR,IHDIR,4)
      |                    1
Warning: Actual argument contains too few elements for dummy argument ‘holl’ (4/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:3141:30:

 3141 |                   CALL MZBOOK(JQPDVS,LDIR,LRZ,-1,'RZ  ',6,6,LREF,2,-1)
      |                              1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3150:33:

 3150 |                      CALL MZPUSH(JQPDVS,LRZ,0,LREF*(NRDS-1),' ')
      |                                 1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5928:52:

 5928 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR) CALL MZSDIV (IXSTOR,-7)
      |                                                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5929:19:

 5929 |       CALL MZCHLS (-7,LHEAD)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5957:19:

 5957 |       CALL MZCHLS (-7,LNEW)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:5969:19:

 5969 |       CALL MZCHLS (-7,LNEW)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1724:53:

 1724 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR)  CALL MZSDIV (IXSTOR,-7)
      |                                                     1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4772:52:

 4772 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR) CALL MZSDIV (IXSTOR,-7)
      |                                                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4008:21:

 4008 |          CALL MZBOOK(JQPDVS,LRIN,LTOP,-7,'RZIN',0,0,LREC+1,2,-1)
      |                     1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4052:42:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2              
......
 4052 |             CALL UCOPYI(IQ(KQSP+LRIN+IS1),IV,NP1)
      |                                          1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:4062:43:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2               
......
 4062 |             CALL UCOPYI(IQ(KQSP+LROUT+IS1),IV,NP1)
      |                                           1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:1032:18:

 1032 |       CALL RZREAD(NWTABI,3,1,1)
      |                  1
Warning: Rank mismatch in argument ‘iv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1040:18:

 1040 |       CALL MZSDIV(IXDIVI,7)
      |                  1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1121:28:

 1121 |          CALL ZSHUNT(IXDIVI,LENTRI,LSUPP,JB,1)
      |                            1
Warning: Rank mismatch in argument ‘lshp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6975:52:

 6975 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR) CALL MZSDIV (IXSTOR,-7)
      |                                                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6976:19:

 6976 |       CALL MZCHLS (-7,LSH)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6989:19:

 6989 |       CALL MZCHLS (-7,LSUP)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7001:19:

 7001 |       CALL MZCHLS (-7,LSUP)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7031:19:

 7031 |    53 CALL MZCHLS (-7,LNEX)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7044:19:

 7044 |       CALL MZCHLS (-7,LNEX)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:1590:19:

 1590 |       CALL MZRELL (LADESV)
      |                   1
Warning: Actual argument contains too few elements for dummy argument ‘mdesv’ (6/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:1664:19:

 1664 |    24 CALL MZCHLN (-7,IQNX)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4192:52:

 4192 |       IF (JBYT(IXSTOR,27,6).NE.JQSTOR) CALL MZSDIV (IXSTOR,-7)
      |                                                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6741:19:

 6741 |       CALL MZCHLS (-7,LINK)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6761:19:

 6761 |       CALL MZCHLS (-7,LINK)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6527:19:

 6527 |       CALL MZCHLN (-7,LN)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6578:19:

 6578 |       CALL MZCHLS (-7,LINK)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:6608:19:

 6608 |       CALL MZCHLS (-7,LINK)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3458:23:

 3458 |       IXSTOR = MZIXCO (IXSTOR+21,22,23,24)
      |                       1
Warning: Rank mismatch in argument ‘ixaa’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3459:19:

 3459 |       CALL MZGARB (IXSTOR, 0)
      |                   1
Warning: Rank mismatch in argument ‘ixgp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3465:18:

 3465 |       J = MZIXCO (21,22,23,24)
      |                  1
Warning: Rank mismatch in argument ‘ixaa’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3466:19:

 3466 |       CALL MZGARB (J, 0)
      |                   1
Warning: Rank mismatch in argument ‘ixgp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3467:19:

 3467 |       CALL MZSDIV (IXSTOR,-7)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4405:19:

 4405 |       CALL MZSDIV (IXIN,-7)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4411:19:

 4411 |    29 CALL MZSDIV (IXIN,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3647:66:

 3647 |                      CALL ZITOH(IQ(KQSP+LCDIR+LKC+K),IQUEST(20+K),1)
      |                                                                  1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3728:58:

 3728 |                      CALL ZITOH(IQUEST(30+J),IQUEST(30+J),1)
      |                                                          1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3744:58:

 3744 |                      CALL ZITOH(IQUEST(40+J),IQUEST(40+J),1)
      |                                                          1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3764:55:

 3764 |                   CALL ZITOH(IQUEST(30+J),IQUEST(30+J),1)
      |                                                       1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3765:55:

 3765 |                   CALL ZITOH(IQUEST(40+J),IQUEST(40+J),1)
      |                                                       1
Warning: Rank mismatch in argument ‘np’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:308:38:

  308 |       IF (N.NE.0)  CALL UCTOH (CHNAME,NQSNAM,4,N)
      |                                      1
......
 1797 |       IF  (NIO.NE.0)  CALL UCTOH (CHIDH,NQID,4,NIO)
      |                                        2
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:413:19:

  413 |       CALL MZLIFT (-7,LSYS,0,2,MMSYSL,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4990:19:

 4990 |       CALL MZCHLN (-7,LN)
      |                   1
Warning: Rank mismatch in argument ‘ixst’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:69:26:

   69 |       CALL UCOPYI (INKEYS,MQKEYS,3)
      |                          1
......
 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:130:19:

  130 |       CALL IZHNUM (IQLETT,NQHOLL,95)
      |                   1
Warning: Actual argument contains too few elements for dummy argument ‘holl’ (26/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:209:38:

  209 |       IF (IPAW(1).NE.0)   CALL MZEBRA(-1)
      |                                      1
Warning: Rank mismatch in argument ‘list’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:211:19:

  211 |       CALL MZSTOR (IXPAWC,'/PAWC/',' ',IFENCE,LMAIN,IPAW(1),IPAW(1),
      |                   1
Warning: Rank mismatch in argument ‘ixstor’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:922:21:

  922 |          CALL MZBOOK(JQPDVS,LRIN,LTOP,-7,'RZIN',0,0,LREC+1,2,-1)
      |                     1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:931:43:

  931 |       IF(IOPTS.NE.0)CALL UCOPYI(IQUEST(20),IQKS,10)
      |                                           1
......
 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2               
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:934:32:

  934 |          CALL UCOPYI(IQUEST(41),IQK,10)
      |                                1
......
 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2    
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:938:21:

  938 |          CALL MZSDIV(IXDIV,1)
      |                     1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:944:20:

  944 |          CALL RZINS(IXDIV,LSUP,JBIAS,LBANK)
      |                    1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:955:24:

  955 |             CALL MZBOOK(IXDIV,LFROM,LSUP,JBIAS,'RZIN',0,0,NDATA,
      |                        1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2581:19:

 2581 |    22 CALL MZSDIV (JDV,4)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2912:19:

 2912 |       CALL MZSDIV (0,-7)
      |                   1
Warning: Rank mismatch in argument ‘ixdivp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2927:35:

 2927 |          CALL MZLINK(JQPDVS,'RZCL',LTOP,LTOP,LFROM)
      |                                   1
Warning: Rank mismatch in argument ‘larea’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2928:22:

 2928 |          CALL MZBOOK (JQPDVS,LRZ0,LQRS,1,'RZ0 ',2,2,36,2,0)
      |                      1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2937:18:

 2937 |       CALL MZBOOK(JQPDVS,LTOP,LQRS,1,'RZ  ',10,9,LRECP,2,0)
      |                  1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2950:24:

 2950 |             CALL MZPUSH(JQPDVS,LRZ0,0,10,' ')
      |                        1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2964:32:

 2964 |       IF(NPUSH.NE.0)CALL MZPUSH(JQPDVS,LTOP,0,NPUSH,'I')
      |                                1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:2974:37:

 1878 |           CALL UCOPYI (NAME,NQID,NIO+5)
      |                            2         
......
 2974 |       CALL UCOPYI(IQ(KQSP+LTOP+KPW1),IHPWD,2)
      |                                     1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:3007:21:

 3007 |          CALL MZBOOK(JQPDVS,LFREE,LTOP,-2,'RZFR',0,0,21,2,0)
      |                     1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3016:34:

 3016 |                CALL MZDROP(JQPDVS,LFREE,' ')
      |                                  1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:3024:21:

 3024 |          CALL MZBOOK(JQPDVS,LUSED,LTOP,-3,'RZUS',0,0,21,2,0)
      |                     1
Warning: Rank mismatch in argument ‘ixp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4700:41:

 4700 |       CALL MZIOCH (IQ(KQSP+LIOD+IXIOD+1),16,CHFORM)
      |                                         1
Warning: Rank mismatch in argument ‘nwiomp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4721:19:

 4721 |    71 CALL MZPUSH (JQPDVS,LID,0,20,'I')
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4723:19:

 4723 |       CALL MZPUSH (JQPDVS,LIX,0,20,'I')
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4726:19:

 4726 |       CALL MZPUSH (JQPDVS,LIOD,0,60,'I')
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4730:19:

 4730 |       CALL MZLIFT (JQPDVS,L,LQFORM,1,MMID,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4731:19:

 4731 |       CALL MZLIFT (JQPDVS,LIX,L,-1,MMIX,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:4733:19:

 4733 |       CALL MZLIFT (JQPDVS,L,LQFORM,-2,MMIO,0)
      |                   1
Warning: Rank mismatch in argument ‘ixdiv’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7175:38:

 7175 |          CALL ZITOH(IQ(KQSP+LCDIR+IH),IHDIR,4)
      |                                      1
Warning: Actual argument contains too few elements for dummy argument ‘iholl’ (4/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:7183:31:

 7183 |             CALL MZDROP(JQPDVS,LCDIR,' ')
      |                               1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7217:19:

 7217 |       CALL MZGARB (0,IXWIPE)
      |                   1
Warning: Rank mismatch in argument ‘ixgp’ at (1) (rank-1 and scalar)
/opt/root_src/misc/minicern/src/zebra.f:7267:23:

 1797 |       IF  (NIO.NE.0)  CALL UCTOH (CHIDH,NQID,4,NIO)
      |                                        2
......
 7267 |       CALL UCTOH(CHDIR,IHDIR,4,NCHD)
      |                       1
Warning: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
/opt/root_src/misc/minicern/src/zebra.f:7268:17:

 7268 |       CALL ZHTOI(IHDIR,IHDIR,4)
      |                 1
Warning: Actual argument contains too few elements for dummy argument ‘holl’ (4/99) at (1)
/opt/root_src/misc/minicern/src/zebra.f:7284:28:

 7284 |          CALL MZDROP(JQPDVS,LTOP,' ')
      |                            1
Warning: Rank mismatch in argument ‘lheadp’ at (1) (rank-1 and scalar)
```

Very rarely (just once) I got this debug stacktrace (2158 in my debug build was at line `NWTAB = IQ(KQS+LSYS+1)`):

```
/opt/root_src/misc/minicern/src/zebra.f:2158:72: runtime error: load of address 0x64d8ffa72fc8 with insufficient space for an object of type 'integer(kind=4)'
0x64d8ffa72fc8: note: pointer points here



/opt/root_src/misc/minicern/src/zebra.f:2158:72: runtime error: load of address 0x64d8ffa72fc8 with insufficient space for an object of type 'integer(kind=4)'
0x64d8ffa72fc8: note: pointer points here
<memory cannot be printed>

 *** Break *** segmentation violation



===========================================================
There was a crash (kSigSegmentationViolation).
This is the entire stack trace of all threads:
===========================================================
#0  0x00007da33a6ea3ea in __GI___wait4 (pid=1162663, stat_loc=stat_loc
entry=0x7fff5a35ae28, options=options
entry=0, usage=usage
entry=0x0) at ../sysdeps/unix/sysv/linux/wait4.c:30
#1  0x00007da33a6ea3ab in __GI___waitpid (pid=<optimized out>, stat_loc=stat_loc
entry=0x7fff5a35ae28, options=options
entry=0) at ./posix/waitpid.c:38
#2  0x00007da33a650bdb in do_system (line=<optimized out>) at ../sysdeps/posix/system.c:171
#3  0x00007da33b366d5a in TUnixSystem::Exec (this=0x617000001200, shellcmd=0x60700026bc80 "/home/user/build/build-root-Desktop-Debug/etc/gdb-backtrace.sh 1162652 1>&2") at /opt/root_src/core/unix/src/TUnixSystem.cxx:2160
#4  0x00007da33b36767e in TUnixSystem::StackTrace (this=0x617000001200) at /opt/root_src/core/unix/src/TUnixSystem.cxx:2451
#5  0x00007da33b36b515 in TUnixSystem::DispatchSignals (this=0x617000001200, sig=kSigSegmentationViolation) at /opt/root_src/core/unix/src/TUnixSystem.cxx:3671
#6  0x00007da33b3626fe in SigHandler (sig=kSigSegmentationViolation) at /opt/root_src/core/unix/src/TUnixSystem.cxx:410
#7  0x00007da33b36b413 in sighandler (sig=11) at /opt/root_src/core/unix/src/TUnixSystem.cxx:3642
#8  <signal handler called>
#9  0x000064d7ff3a5020 in mzlink (ixstor=0, chname=..., larea=..., lref=..., lrefl=..., _chname=8) at /opt/root_src/misc/minicern/src/zebra.f:2158
#10 0x000064d7ff34b6d1 in hlimit (limit=4000000) at /opt/root_src/misc/minicern/src/hbook.f:206
#11 0x000064d7ff2f39eb in main (argc=2, argv=0x7fff5a35e838) at /opt/root_src/main/src/h2root.cxx:311
===========================================================


The lines below might hint at the cause of the crash. If you see question
marks as part of the stack trace, try to recompile with debugging information
enabled and export CLING_DEBUG=1 environment variable before running.
You may get help by asking at the ROOT forum https://root.cern/forum
preferably using the command (.forum bug) in the ROOT prompt.
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs or (preferably) using the command (.gh bug) in
the ROOT prompt. Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#9  0x000064d7ff3a5020 in mzlink (ixstor=0, chname=..., larea=..., lref=..., lrefl=..., _chname=8) at /opt/root_src/misc/minicern/src/zebra.f:2158
#10 0x000064d7ff34b6d1 in hlimit (limit=4000000) at /opt/root_src/misc/minicern/src/hbook.f:206
#11 0x000064d7ff2f39eb in main (argc=2, argv=0x7fff5a35e838) at /opt/root_src/main/src/h2root.cxx:311
===========================================================


==1162752==Processing thread 1162652.
==1162752==Stack at 0x7fff59b61000-0x7fff5a361000 (SP = 0x7fff5a35cea8).
==1162752==TLS at 0x7da33ddf1000-0x7da33ddf7680.
```

Mac15 CI with -O0 removed but -g active:
```
    *** Break *** bus error
   Converting directory //example
  [/usr/lib/system/libsystem_platform.dylib] _sigtramp (no debug info)
  [/Users/sftnight/ROOT-CI/build/bin/h2root] hgive_ /Users/sftnight/ROOT-CI/src/misc/minicern/src/hbook.f:802
  [/Users/sftnight/ROOT-CI/build/bin/h2root] convert_1d(int) /Users/sftnight/ROOT-CI/src/main/src/h2root.cxx:473
  [/Users/sftnight/ROOT-CI/build/bin/h2root] convert_directory(char const*) /Users/sftnight/ROOT-CI/src/main/src/h2root.cxx:0
  [/Users/sftnight/ROOT-CI/build/bin/h2root] main /Users/sftnight/ROOT-CI/src/main/src/h2root.cxx:344
```

From https://github.com/root-project/root/pull/19236
Happens maybe only in some gfortran versions
Maybe related: https://www.tek-tips.com/threads/string-copy.1609892/post-6335242
in mac13 and mac14 CI, the last bytes of chopt were being set to garbage memory, which later led to failures due to wrong options being set. chopt was being set to "px exam"
where choptt was "px " and "exam" was the first chars of the parameter PASSCHAR("example") on the C side in h2root.cxx
mac14 still fails (with another error), but mac13 h2root works now.
mac13: there is still the error in rpath for two tests.
mac15: failures due to runners having different gfortran versions?


Other cases:

```
  Ubuntu 22 imt=off Debug build, sometimes:
  
  cd /github/home/ROOT-CI/build/roottest/root/hist
  /usr/bin/timeout -s USR2 270s h2root mb4i1.hbook
  -- BEGIN TEST OUTPUT --
  
   *** Break *** segmentation violation
   Generating stack trace...
   0x000055aeff2de9c0 in mzlink_ + 0x1ac from h2root
   0x000055aeff2ceda9 in hlimit_ + 0xbd from h2root
   0x000055aeff2bd7ab in main + 0x2d7 from h2root
   0x00007f255ac2ed90 in <unknown> from /lib/x86_64-linux-gnu/libc.so.6
   0x00007f255ac2ee40 in __libc_start_main + 0x80 from /lib/x86_64-linux-gnu/libc.so.6
   0x000055aeff2bd405 in _start + 0x25 from h2root
  
  -- END TEST OUTPUT --
  CMake Error at /github/home/ROOT-CI/build/RootTestDriver.cmake:186 (message):
    got exit code 139 but expected 0
 
 alma9 march=native RelWithDebInfo build   
(sometimes)
    
  -- TEST COMMAND -- 
  cd /github/home/ROOT-CI/build/roottest/root/hist
  /usr/bin/timeout -s USR2 270s h2root mb4i1.hbook
  -- BEGIN TEST OUTPUT --
   RZIODO. Error at record =    2 LUN =    10 IOSTAT =  5001
   >>>>>> CALL RZEND(CHDIR)
   Cannot open fileHROPEN           0
   Error on hropen was 101 
  Error cannot open input file: mb4i1.hbook
  
  -- END TEST OUTPUT --
  CMake Error at /github/home/ROOT-CI/build/RootTestDriver.cmake:186 (message):
    got exit code 1 but expected 0
```

```
  /github/home/ROOT-CI/src/misc/minicern/src/zebra.f:6746:72: runtime error: load of address 0x56353fb115cc with insufficient space for an object of type 'integer(kind=4)'
  0x56353fb115cc: note: pointer points here
    24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
                ^ 
  /github/home/ROOT-CI/src/misc/minicern/src/zebra.f:6747:72: runtime error: load of address 0x56353fb115cc with insufficient space for an object of type 'integer(kind=4)'
  0x56353fb115cc: note: pointer points here
    24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
                ^ 
  /github/home/ROOT-CI/src/misc/minicern/src/zebra.f:6753:72: runtime error: store to address 0x56353fb115cc with insufficient space for an object of type 'integer(kind=4)'
  0x56353fb115cc: note: pointer points here
    24 02 00 00 18 2f 0f 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00
                ^ 
```    

alma9 arm64 without -O0, got also (https://github.com/root-project/root/pull/20538 and https://github.com/root-project/root/actions/runs/19712084490/job/56475155682?pr=20538)

```
   *** Break *** segmentation violation
   Generating stack trace...
   0x000000000040a7ec in hrfile_ at /github/home/ROOT-CI/src/misc/minicern/src/hbook.f:352 from h2root
   0x000000000040ff88 in hropen_ at /github/home/ROOT-CI/src/misc/minicern/src/hbook.f:277 from h2root
   0x0000000000404d70 in main at /github/home/ROOT-CI/src/main/src/h2root.cxx:333 from h2root
   0x0000ffffb36f8340 in <unknown> from /lib64/libc.so.6
   0x0000ffffb36f8418 in __libc_start_main at :? from /lib64/libc.so.6
   0x00000000004050b0 in _start + 0x30 from h2root
```


If I turn on Wall Wextra warnings via set_target_properties(minicern PROPERTIES COMPILE_FLAGS "-Wall -Wextra"), I see (https://github.com/root-project/root/actions/runs/20062341758/job/57542127199?pr=20538)

several warnings of the type:

```
 Warning: Array reference at (1) out of bounds (8729 > 8704) in loop beginning at (2) [-Wdo-subscript]
  /github/home/ROOT-CI/src/misc/minicern/src/zebra.f:574:45:

   /github/home/ROOT-CI/src/misc/minicern/src/hbook.f:2514:72:
  
   2514 |       BSLASH='\\'
        |                                                                        1
  Warning: CHARACTER expression will be truncated in assignment (1/2) at (1) [-Wcharacter-truncation]

```

### Reproducer

Run `h2root roottest/root/hist/mb4i1.hbook`, optionally enabling the address sanitizer in main/CMakeLists.txt or changing optimizer flag to -O3 or removing -W flag.

On the CI, run the optimized single-test CI script: https://github.com/root-project/root/pull/19191

### ROOT version

```
   ------------------------------------------------------------------
  | Welcome to ROOT 6.37.01                        https://root.cern |
  | (c) 1995-2025, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jul 05 2025, 18:26:49                 |
  | From heads/master@v6-31-01-13222-g951be70a61                     |
  | With g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0                   |
  | Try '.help'/'.?', '.demo', '.license', '.credits', '.quit'/'.q'  |
   ------------------------------------------------------------------
```

### Installation method

From source

### Operating system

Ubuntu22

### Additional context

https://root-forum.cern.ch/t/bug-in-h2root/64759

https://github.com/root-project/root/pull/19191
https://github.com/root-project/root/pull/19184
https://github.com/root-project/root/pull/19236
https://github.com/root-project/root/pull/19200
https://github.com/root-project/root/pull/15915/
https://github.com/root-project/root/issues/14155
https://its.cern.ch/jira/browse/ROOT-9179

--- a/main/src/h2root.cxx
+++ b/main/src/h2root.cxx
@@ -134,11 +134,7 @@ void MAIN__() {}
 
 // As recommended in
 // https://gcc.gnu.org/onlinedocs/gfortran/Argument-passing-conventions.html
-#if __GNUC__ > 7
-typedef size_t fortran_charlen_t;
-#else
-typedef int fortran_charlen_t;
-#endif
+using fortran_charlen_t = size_t;
 
 #else
 # define hlimit  HLIMIT
@@ -329,11 +325,11 @@ int main(int argc, char **argv)
 
    int lun = 10;
 #ifndef WIN32
-   // "px" is a string being changed to "pxc" in the fortran side; since it's a temporary on C's side,
-   // we need a buffer of at least 3 chars on Cs side, too. Predefine it as a extra variable "px " since that way
-   // the space will be replaced by 'c' on the preallocated memory, instead of appending a new char (new memory) to "px".
-   auto opt = PASSCHAR("px ");
-   hropen(lun,PASSCHAR("example"),PASSCHAR(file_in),opt,record_size,ier,7,strlen(file_in),2);
+   // Reminder: Argument passing convention is that string lengths go as hidden arguments
+   // at the end of the argument list.
+   constexpr auto chdir = "example";
+   constexpr auto opt = "px";
+   hropen(lun, chdir, file_in, opt, record_size, ier, strlen(chdir), strlen(file_in), strlen(opt));
 #else
    hropen(lun,PASSCHAR("example"),PASSCHAR(file_in),PASSCHAR("px"),record_size,ier);
 #endif

